testing/endtoend/component_handler_test.go

```diff
@@ -137,23 +138,47 @@ func (c *componentHandler) setup() {
 	if config.TestCheckpointSync {
 		appendDebugEndpoints(config)
 	}
-	// Proxies
-	proxies := eth1.NewProxySet()
-	g.Go(func() error {
-		if err := helpers.ComponentsStarted(ctx, []e2etypes.ComponentRunner{eth1Nodes}); err != nil {
-			return errors.Wrap(err, "proxies require execution nodes to run")
-		}
-		if err := proxies.Start(ctx); err != nil {
-			return errors.Wrap(err, "failed to start proxies")
-		}
-		return nil
-	})
-	c.eth1Proxy = proxies
+
+	var builders *components.BuilderSet
+	var proxies *eth1.ProxySet
+	if config.UseBuilder {
+		// Builder
+		builders = components.NewBuilderSet()
+		g.Go(func() error {
+			if err := helpers.ComponentsStarted(ctx, []e2etypes.ComponentRunner{eth1Nodes}); err != nil {
+				return errors.Wrap(err, "builders require execution nodes to run")
+			}
+			if err := builders.Start(ctx); err != nil {
+				return errors.Wrap(err, "failed to start builders")
+			}
+			return nil
+		})
+		c.builders = builders
+	} else {
+		// Proxies
+		proxies = eth1.NewProxySet()
+		g.Go(func() error {
+			if err := helpers.ComponentsStarted(ctx, []e2etypes.ComponentRunner{eth1Nodes}); err != nil {
+				return errors.Wrap(err, "proxies require execution nodes to run")
```

Comment:
how does the proxy work? it sits between our beacon node and a running execution client? and we are replacing it? I thought the builder should run along side the eth1Node so I thought there would be essentially 2 proxies 1 for builders and 1 for eth1 node. in case we have a scenerio test for fall back.

Reply:
Yes, the only way the builder will work is if it functions as a proxy. It needs the FCU calls from the beacon node in order to know the payload id of which payloads to retrieve. Which is why it replaces the proxies outright here